docs/complex-types/methods.md

```diff
@@ -0,0 +1,41 @@
+# methods
+
+In this section we will explore 🚀 all nooks and crannies behind methods types.
+
+
+
+Diving into `methods.ts` we have these three imports
+
+```ts
+import { EntitySchema } from "./data" // Record<string, any>
+import { SyncKey } from "./sync" // Opaque<string, "sync-key">
+import { Except } from "type-fest" // Create a type from an object type without certain keys.
+```
+
+## Methods
+
+```ts
+export type Methods<TSchema extends EntitySchema> = Record<
+  string, // keys
+  (this: TSchema, ...args: any[]) => any // values
+>
+```
+
+`Methods<TSchema>`: This is a generic type called `Methods`. It takes a type parameter `TSchema`, which is expected to extend the `EntitySchema`.<br/>
+This means that `Methods` is a reusable type where we can use it for different EntityType. (dunno if I should explain this @TODO)
```

Comment:
Skup się najpierw na celu jaki realizuje dany typ - w tym przykładzie `Methods` stanowią strukturę dla deklaracji metod w modelu Entity.

Dopiero w kolejnej sekcji jak ten typ jest zbudowany - ale skupiając się  tylko na najtrudniejszych kawałkach.

Reply:
Wydaje się, że rozpykane to zostało :smile: